applications/FluidDynamicsApplication/custom_elements/embedded_fluid_element.cpp

```diff
@@ -231,14 +242,396 @@ void EmbeddedFluidElement<TBaseElement>::NormalizeInterfaceNormals(
+        const bounded_matrix<double, StrainSize, LocalSize> aux_matrix_CB = prod(rData.C, B_matrix);
+        const bounded_matrix<double, Dim, LocalSize> aux_matrix_ACB = prod(voigt_normal_proj_matrix, aux_matrix_CB);
+        const bounded_matrix<double, LocalSize, LocalSize> aux_matrix_BtransAtransPtanACB = prod(aux_matrix_BtransAtransPtan, aux_matrix_ACB);
+
+        // Contribution coming from the traction vector tangencial component
+        noalias(aux_LHS_1) -= adjoint_consistency*nitsche_coefs.first*weight*aux_matrix_BtransAtransPtanACB;
+
+        // Contribution coming from the shear force generated by the velocity jump
+        noalias(aux_LHS_2) -= adjoint_consistency*nitsche_coefs.second*weight*prod(aux_matrix_BtransAtransPtan, N_mat);
+
+    }
+
+    // LHS outside Nitche contribution assembly
+    noalias(rLHS) += aux_LHS_1;
+    noalias(rLHS) += aux_LHS_2;
+
+    // RHS outside Nitche contribution assembly
+    // If level set velocity is not 0, add its contribution to the RHS
+    if (this->Has(EMBEDDED_VELOCITY)) {
+        const array_1d<double, 3 >& embedded_vel = this->GetValue(EMBEDDED_VELOCITY);
+        array_1d<double, LocalSize> embedded_vel_exp = ZeroVector(LocalSize);
+
+        for (unsigned int i = 0; i < NumNodes; ++i) {
+            for (unsigned int comp = 0; comp < Dim; ++comp) {
+                embedded_vel_exp(i*BlockSize + comp) = embedded_vel(comp);
+            }
+        }
+
+        noalias(rRHS) += prod(aux_LHS_2, embedded_vel_exp);
+    }
+
+    // Note that since we work with a residualbased formulation, the RHS is f_gamma - LHS*prev_sol
+    noalias(rRHS) -= prod(aux_LHS_1, values);
+    noalias(rRHS) -= prod(aux_LHS_2, values);
+
+}
+
+template <class TBaseElement>
+double EmbeddedFluidElement<TBaseElement>::ComputeSlipNormalPenaltyCoefficient(
+    const EmbeddedElementData& rData) const {
+
+    // Compute the effective viscosity as the average of the lower diagonal constitutive tensor
+    // TODO: TO BE OBTAINED FROM THE CLAW ONCE JORDI FINISHES HIS IMPLEMENTATION
+    double eff_mu = rData.DynamicViscosity;
+
+    // Compute the element average velocity norm
+    double v_norm = 0.0;
+    for (unsigned int comp = 0; comp < Dim; ++comp){
+        double aux_vel = 0.0;
+        for (unsigned int j = 0; j < NumNodes; ++j){
+            aux_vel += rData.Velocity(j,comp);
+        }
+        aux_vel /= NumNodes;
+        v_norm += aux_vel*aux_vel;
+    }
+    v_norm = std::sqrt(v_norm);
+
+    // Compute the element average density
+    double avg_rho = rData.Density;
+
+    // Compute the Nitsche coefficient (including the Winter stabilization term)
+    const double h = ElementSizeCalculator<Dim,NumNodes>::GradientsElementSize(rData.DN_DX);
+    const double penalty = 1.0/10.0; // TODO: SHOULD WE EXPORT THIS TO THE USER SIDE?¿?¿
+    const double cons_coef = (eff_mu + eff_mu + avg_rho*v_norm*h + avg_rho*h*h/rData.DeltaTime)/(h*penalty);
```

Comment:
should `eff_mu` be added twice? (no idea, just asking in case it is a copy-paste error)